### ISSUE TITLE
[REFACTOR] FIX PUBLISH SUBSCRIBE

### ISSUE THREAD

**Issue by DerekTBrown** [learnlinux/tuxlab-app#73]
You **CANNOT** simply publish an entire collection (https://github.com/learnlinux/tuxlab-app/blob/de9bc0d54b9d7c504d4ded9a9be728b921e6df90/collections/courses.ts#L7). What this will do is send a copy of all of the data in the courses database to the user.  This is **REALLY REALLY REALLY** bad.  Please revisit publish-subscribe, particularly concepts like [pagination](https://guide.meteor.com/data-loading.html#pagination) and [authorization](https://guide.meteor.com/data-loading.html#complex-auth).  

In addition, remember that you should be limiting what data is sent to the client- they do not need a copy of the labfile, or tasks (this should be hidden for them until they complete them).  I have made this a lot easier by splitting courses and labs into separate databases.

**Comment by sandershihacker** [comment]
Done with course_records. Publishing those that include the current userId, and none if no user is logged in.

**Comment by sandershihacker** [comment]
Done with courses. Publishing those that are registered by current user.

**Comment by DerekTBrown** [comment]
Okay- so I did some performance testing this morning, and I believe I have found the best way to do it.  There is a reactive publication library that automatically handles updates to the code, and given there is a 1:1 relationship between our updates (enforced by the schema), this should work the best. (https://github.com/peerlibrary/meteor-reactive-publish#examples). I have written you an example of what this looks like in the courses.ts file in the pubsub branch.  

Logically, there are three types of publish statements we need:

1) Automatically publishes courses, labs, and course records the user has access to.  This should be done reactively.

2) Manual publish requests based on routing.  For example, if a global admin goes to a page, they should get all of the data (http://meteortips.com/second-meteor-tutorial/publish-subscribe/)

3) search results on the explore page
